providers/base/bin/camera_quality_test.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+#
+# This file is part of Checkbox.
+#
+# Copyright 2008-2023 Canonical Ltd.
+# Written by:
+#   Fernando Bravo <fernando.bravo.hernandez@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import argparse
+import time
+import sys
+
+import cv2
+from numpy import isnan
+
+from checkbox_support.vendor.brisque.brisque import BRISQUE
+from tempfile import NamedTemporaryFile
+
+THRESHOLD = 60
+
+
+def get_image_from_device(device: str):
+    # Set the video device
+    index = int(device.replace("video", ""))
+    cam = cv2.VideoCapture(index)
+
+    if not cam.isOpened():
+        print("Cannot open the selected device", file=sys.stderr)
```

Comment:
I agree with logging the error. I am not so sure about directly rising SystemExit. In the future, we may want to try alternative capture methodsif openCV fails instread of rising SystemExit.

Reply:
Then raise another exception, avoid returning None, it is not descriptive of what went wrong and if unhandled may lead to weird consequences that are hard to pin down. It is better to raise an error so that if you don't handle it will cause a traceback